### PR TITLE
Adds tutorials for Gromacs, Openfoam, & WRF that walkthrough running real workloads

### PR DESCRIPTION
**Pull request by nick-stroud**
This PR adds 3 application specific tutorials. See docs/tutorials/README.md update for more info.

Tested:
- Ran each blueprint in its current form (without cache)
  - Create/deploy/run workload
- Spot checked tutorial in open in cloud shell

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: docs/tutorials/openfoam/spack-openfoam.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+In this tutorial you will use the HPC Toolkit to:
+
+* Deploy a [Slurm](https://github.com/SchedMD/slurm-gcp#readme) HPC cluster on
+  Google Cloud
+* Use [Spack](https://spack.io/) to install the OpenFOAM application and all of
+  its dependencies
+* Run a [OpenFOAM](https://www.openfoam.com/) job on your newly provisioned
+  cluster
+* Tear down the cluster
+
```
Comment (cboneti) [inline docs/tutorials/openfoam/spack-openfoam.md:15]: Here, I would maybe add the following lines at the bottom.

"Deploying the cluster and installing the applications dependencies may take 2 hours and 34 minutes if no spack cache is provided. With a complete spack cache, this time can be reduce to 6 minutes." (please check this time).

File: docs/tutorials/wrfv3/spack-wrfv3.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+In this tutorial you will use the HPC Toolkit to:
+
+* Deploy a [Slurm](https://github.com/SchedMD/slurm-gcp#readme) HPC cluster on
+  Google Cloud
+* Use [Spack](https://spack.io/) to install the Weather Research and Forecasting (WRF) Model application and all of
+  its dependencies
+* Run a [Weather Research and Forecasting (WRF) Model](https://www.mmm.ucar.edu/weather-research-and-forecasting-model) job on your newly provisioned
+  cluster
+* Tear down the cluster
+
```
Comment (cboneti) [inline docs/tutorials/wrfv3/spack-wrfv3.md:15]: Here, I would maybe add the following lines at the bottom.

"Deploying the cluster and installing the applications dependencies may take 1.5 hours if no spack cache is provided. With a complete spack cache, this time can be reduce to 6 minutes." (please check this time).

**Review by cboneti — COMMENTED** [review]
Looks good. 
Added a few minor comments.

File: docs/tutorials/gromacs/spack-gromacs.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+In this tutorial you will use the HPC Toolkit to:
+
+* Deploy a [Slurm](https://github.com/SchedMD/slurm-gcp#readme) HPC cluster on
+  Google Cloud
+* Use [Spack](https://spack.io/) to install the Gromacs application and all of
+  its dependencies
+* Run a [Gromacs](https://www.gromacs.org/) job on your newly provisioned
+  cluster
+* Tear down the cluster
+
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/tutorials/gromacs/spack-gromacs.md:15]: Done. PTAL as final was slightly different that what you suggested.

File: docs/tutorials/openfoam/spack-openfoam.md (hunk 1 above)
Comment (nick-stroud) [reply docs/tutorials/openfoam/spack-openfoam.md:15]: Done. PTAL as final was slightly different that what you suggested.

File: docs/tutorials/wrfv3/spack-wrfv3.md (hunk 2 above)
Comment (nick-stroud) [reply docs/tutorials/wrfv3/spack-wrfv3.md:15]: Done. PTAL as final was slightly different that what you suggested.

File: docs/tutorials/gromacs/spack-gromacs.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+Once you have selected a project, click START.
+
+## Enable APIs & Permissions
+
+In a new Google Cloud project there are several apis that must be enabled to
+deploy your HPC cluster. These will be caught when you perform `terraform apply`
+but you can save time by enabling them now by running:
+
+<!-- Tried the native way to do this and it timed out. Leaving comment here for future reference. -->
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/tutorials/gromacs/spack-gromacs.md]: Good call, it is working now, I updated this and will probably hit all other tutorials in a separate pr.

[the remark this answers is not in the document]
Comment (nick-stroud) [reply docs/tutorials/gromacs/spack-gromacs.md]: other tutorials handled in #473.

File: docs/tutorials/gromacs/spack-gromacs.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cboneti) [reply docs/tutorials/gromacs/spack-gromacs.md:15]: LGTM

**Review by cboneti — APPROVED** [review]

**Pull request merged**